packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -507,10 +510,22 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
             return;
         }
 
+        // Fetch the resolution checkpoint from storage.
+        ResolutionCheckpoint memory checkpoint = resolutionCheckpoints[_claimIndex];
+
+        // If the checkpoint does not currently exist, initialize the current left most position as max u128.
+        if (!checkpoint.initialCheckpointComplete) {
+            checkpoint.leftmostPosition = Position.wrap(type(uint128).max);
+            checkpoint.initialCheckpointComplete = true;
+
+            // If `_numToResolve == 0`, assume that we can check all child subgames in this one callframe.
+            if (_numToResolve == 0) _numToResolve = challengeIndicesLen;
+        }
+
         // Assume parent is honest until proven otherwise
-        address countered = address(0);
-        Position leftmostCounter = Position.wrap(type(uint128).max);
-        for (uint256 i = 0; i < challengeIndicesLen; ++i) {
+        uint256 lastToResolve = checkpoint.subgameIndex + _numToResolve;
+        uint256 finalCursor = lastToResolve > challengeIndicesLen ? challengeIndicesLen : lastToResolve;
+        for (uint256 i = checkpoint.subgameIndex; i < finalCursor; ++i) {
```

Comment:
How come we can do a pre-increment here? Shouldn't we need to start at the subgame index since we set it to the final cursor below and the loop conditional performs a non-inclusive comparison?

Reply:
Ah the pre-increment doesn't actually increment the initial `i` value - this loop does start at the initial subgame index. Will change it tho for readability, not sure why it's done like that rn.